BUILD

```diff
@@ -39,6 +39,23 @@ java_library(
     ],
 )
 
+load("//tools:jarjar.bzl", "jarjar_library")
+
+genrule(
+    name = "rules_file",
+    outs = ["rules_file.txt"],
+    cmd = "echo \"rule com.google.auto.common.** dagger.shaded.auto.common.@1\" > $@",
+)
+
+jarjar_library(
+    name = "shaded_compiler",
+    rules_file = ":rules_file.txt",
+    deps = [
+        "//compiler",
+        "@com_google_auto_auto_common//jar",
```

Comment:
I only see auto-common being shaded here: https://github.com/google/dagger/blob/e3d8bf2a25ad83113bb19601982e3ed3312911ff/compiler/pom.xml#L195-L200

I don't think there's anything wrong with shading javapoet, but if no one has needed it it's probably not necessary?